src/shared/watchdog.c

```diff
@@ -17,8 +17,36 @@
 static int watchdog_fd = -1;
 static char *watchdog_device = NULL;
 static usec_t watchdog_timeout = USEC_INFINITY;
+static usec_t watchdog_pretimeout = USEC_INFINITY;
 static usec_t watchdog_last_ping = USEC_INFINITY;
 
+static int update_pretimeout(void) {
+        int r;
+
+        if (watchdog_fd < 0)
+                return 0;
+
+        if (watchdog_pretimeout == USEC_INFINITY || watchdog_timeout == USEC_INFINITY)
+                return 0;
+        else {
+                char buf[FORMAT_TIMESPAN_MAX];
+                int sec = (int) DIV_ROUND_UP(watchdog_pretimeout, USEC_PER_SEC);
```

Comment:
please add overflow handling here... maybe clamp to INT_MAX if larger?

Reply:
done.